service/s3/s3crypto/authentication_mode_test.go

```diff
@@ -0,0 +1,14 @@
+package s3crypto_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aws/aws-sdk-go/service/s3/s3crypto"
+)
+
+func TestAuthenticationModeCipherName(t *testing.T) {
+	mode := s3crypto.Authentication(&s3crypto.SymmetricKeyProvider{})
+	assert.Equal(t, mode.GetCipherName(), "AES/GCM/NoPadding")
```

Comment:
Nit: the order of the equal parameters are reversed. The "expected" should be second, and actual is third.